source/deployment/sso-saml-adfs.md

```diff
@@ -0,0 +1,199 @@
+
+30) Click `Save`.
+
+You’re done! If you’d like to confirm SAML SSO is successfully enabled, switch your System Administrator account from email to SAML-based authentication via **Account Settings > General > Sign-in Method > Switch to SAML SSO** and sign in with your SAML credentials to complete the switch.
+
+It is also recommended to post an announcement about how the migration will work to users.
+
+You may also configure SAML for ADFS by editing `config.json`. Before starting the Mattermost server, edit `config.json` to enable SAML based on [SAML configuration settings](http://docs.mattermost.com/administration/config-settings.html#saml-enterprise). You must restart Mattermost server for the changes to take effect.
+
+#### Troubleshooting
+
+The following are troubleshooting suggestions on common error messages and issues. 
+
+##### 1. System Administrator locks themselves out of the system
+
+If the System Administrator is locked out of the system during SAML configuration process, they can set an existing account to System Administrator using [a commandline tool](http://docs.mattermost.com/deployment/on-boarding.html#creating-system-administrator-account-from-commandline). 
+
+##### 2. Received error message: `An account with that username already exists. Please contact your Administrator.`
+
+This usually means an existing account has another authentication method enabled. If so, the user should sign in using that method (such as email and password), then change their sign-in method to SAML via **Account Settings > Security > Sign-in method**.
+
+This error message can also be received if the `Username Attribute` of their SAML credentials is incorrect. If so, the user can update the attribute at their identity provider (for instance, back to the old value if it had been previously updated). 
+
+##### 3. Received error message: `An account with that email already exists. Please contact your Administrator.`
+
+This usually means an existing account has another authentication method enabled. If so, the user should sign in using that method (such as email and password), then change their sign-in method to SAML via **Account Settings > Security > Sign-in method**.
+
+This error message can also be received if the `Email Attribute` of their SAML credentials is incorrect. If so, the user can update the attribute at their identity provider (for instance, back to the old value if it had been previously updated).
+
+##### 4. Unable to switch to SAML authentication successfully
+
+First, ensure you have installed the [XML Security Library](https://www.aleksey.com/xmlsec/download.html) on your Mattermost instance and that **it is available in your** `PATH`.
+
+Second, ensure you have completed each step in our guides for [configuring SAML with Okta](http://docs.mattermost.com/deployment/sso-saml-okta.html) or for [configuring SAML with Microsoft ADFS](http://docs.mattermost.com/deployment/sso-saml-adfs.html).
```

Comment:
remove Okta note; add this ADFS troubleshooting guide: https://technet.microsoft.com/en-us/library/cc776135(v=ws.10).aspx